### PR TITLE
added changes to the ground plane calibration (removed NaNs)

### PR DESCRIPTION
**Pull request by narora1**

### CODE REVIEW

File: robot_calibration/include/robot_calibration/camera_info.h [hunk 1]
```diff
@@ -43,12 +43,6 @@ enum {CAMERA_PARAMS_CX_INDEX=0,
       CAMERA_PARAMS_Z_SCALE_INDEX=4,
       CAMERA_PARAMS_Z_OFFSET_INDEX=5};
 
-enum {CAMERA_INFO_D_1=0,
```
Comment (mikeferguson) [inline robot_calibration/include/robot_calibration/camera_info.h]: we should not remove these (we'll need them later)

File: robot_calibration/CMakeLists.txt [hunk 2]
```diff
@@ -6,7 +6,6 @@ if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
 endif()
 
 find_package(Boost REQUIRED system thread)
-
```
Comment (mikeferguson) [inline robot_calibration/CMakeLists.txt]: let's restore this newline

File: robot_calibration/src/ground_plane_finder.cpp [hunk 3]
```diff
@@ -120,13 +132,12 @@ bool GroundPlaneFinder::find(robot_calibration_msgs::CalibrationData * msg)
...
-  msg->observations[1].sensor_name = "ground";
+  msg->observations[1].sensor_name = chain_sensor_name_;
   msg->observations[1].features.resize(points_total);
 
-  int step = cloud_.width/points_total;
-
+  size_t step = cloud_.width/points_total;
   size_t k = 0;
 
   for (size_t i = step ; i < cloud_.width ; i = i+step)
```
Comment (mikeferguson) [inline robot_calibration/src/ground_plane_finder.cpp]: does not conform to style guide.

File: robot_calibration/src/led_finder.cpp [hunk 4]
```diff
@@ -59,7 +59,7 @@ LedFinder::LedFinder(ros::NodeHandle& nh) :
                              1,
                              &LedFinder::cameraCallback,
                              this);
-
+
```
Comment (mikeferguson) [inline robot_calibration/src/led_finder.cpp]: this line should not be changed

File: robot_calibration/src/optimizer.cpp [hunk 5]
```diff
@@ -177,11 +177,11 @@ int Optimizer::optimize(OptimizationParams& params,
       {
         std::string camera_name = static_cast<std::string>(params.error_blocks[j].params["camera"]);
         std::string ground_name = static_cast<std::string>(params.error_blocks[j].params["ground"]);
-
+
```
Comment (mikeferguson) [inline robot_calibration/src/optimizer.cpp]: this line (and the one below) should not be changed

File: robot_calibration/src/ground_plane_finder.cpp [hunk 6]
```diff
@@ -144,7 +154,7 @@ bool GroundPlaneFinder::find(robot_calibration_msgs::CalibrationData * msg)
     world.point.z = 0;
 
     // Get 3d point
-    int index = (int)points[i].x;
+    int index =  (int)(points[i].x);
```
Comment (mikeferguson) [inline robot_calibration/src/ground_plane_finder.cpp]: this should be a C++ style cast "static_cast<int>(points[i].x)"

File: robot_calibration/src/ground_plane_finder.cpp [hunk 7]
```diff
@@ -102,6 +89,31 @@ bool GroundPlaneFinder::find(robot_calibration_msgs::CalibrationData * msg)
...
+  for (size_t i = 0; i < num_points; i++)
+  {
+    geometry_msgs::Point p;
+    p.x = (xyz + i)[X];
+    p.y = (xyz + i)[Y];
+    p.z = (xyz + i)[Z];
+
+    if (!std::isfinite(p.x) || !std::isfinite(p.y) || !std::isfinite(p.z))
+      continue;
+    (iter+j)[X] = (xyz+i)[X];
```
Comment (mikeferguson) [inline robot_calibration/src/ground_plane_finder.cpp]: should be a space before and after each plus (this line and 2 more)

**Comment by mikeferguson** [comment]
@narora1 -- mainly style issues, see comments above.

File: robot_calibration/include/robot_calibration/camera_info.h (hunk 1 above)
Comment (narora1) [inline robot_calibration/include/robot_calibration/camera_info.h]: this is fixed but above the block as it makes more sense there (kinda uniform)

File: robot_calibration/src/optimizer.cpp [hunk 8]
```diff
@@ -181,7 +181,7 @@ int Optimizer::optimize(OptimizationParams& params,
         // Check that this sample has the required features/observations
         if (!hasSensor(data[i], camera_name) || !hasSensor(data[i], ground_name))
           continue;
-
+
```
Comment (mikeferguson) [inline robot_calibration/src/optimizer.cpp]: this still needs to be reversed

Comment (narora1) [inline robot_calibration/src/optimizer.cpp]: fixed

**Pull request merged**